client/src/lib/TinyMCEActionRegistrar.js

```diff
@@ -18,26 +28,60 @@ class TinyMCEActionRegistrar {
    *
    * @param {String} menu Name of top level menu item
    * @param {Object} action Menu action option, with custom "priority" property
+   * @param {Boolean} configId Identifier of which config this action should be registered to
    * @return {TinyMCEActionRegistrar}
    */
-  addAction(menu, action) {
+  addAction(menu, action, configId) {
     const priority = action.priority || 50;
-    this.actions[menu] = this.getActions(menu).concat([{ ...action, priority }]);
+    const name = createIdentifier(configId, menu);
+    const actions = this.getActions(menu, configId, true);
+    // If the action is not already registered either globally or within this config
+    if (!actions.find((registeredAction) => action.text === registeredAction.text)) {
+      this.actions[name] = [
+        ...this.getActions(menu, configId, false),
+        { ...action, priority },
+      ];
+    }
     return this;
   }
 
   /**
-   * Get list of actions for this menu, sorted by first by priority and then text
+   * Get list of actions for this menu
    *
    * @param {String} menu
+   * @param {String} configId Identifier of which config
+   * @param {Boolean} includingGlobal Whether to include global actions
+   * not registered to a specific config
    * @return {Array}
    */
-  getActions(menu) {
-    return this.actions[menu] || [];
+  getActions(menu, configId, includingGlobal = true) {
+    // Include global actions
+    // - if registering a global action ie. without a specific config id
+    // - by default or if set to true
+    let actions = (!configId || includingGlobal) && this.actions[menu]
+      ? this.actions[menu]
+      : [];
+    const name = createIdentifier(configId, menu);
+    if (configId && this.actions[name]) {
+      actions = [
+        ...actions,
+        ...this.actions[name],
+      ];
+    }
+    return actions;
   }
 
-  getSortedActions(menu) {
-    const actions = this.getActions(menu);
+  /**
+   * Get list of actions for this menu, sorted by first by priority and then text
+   *
+   * @param {String} menu
+   * @param {Bool} configId Identifier of which config
```

Comment:
This isn't a bool. Also would be good to be consistent with the types (Bool vs Boolean), and also note that I think JSDoc generally uses lowercase for scalar types, other than Object and Array which seem to be used in both uppercased and lowercase in examples